design-proposals/instancetype.kubevirt.io/expand-during-submission.md

```diff
@@ -0,0 +1,105 @@
+# Overview
+
+The initial design of instance types and preferences captured a point in time copy of the original resource for future use by the VirtualMachine to ensure we always generate the same VirtualMachineInstance at runtime.
+
+This additionally allows users to easily switch between classes, sizes and generations of these resources at some point in the future. However this flexibility comes at the cost of complexity within KubeVirt itself especially handling more advanced VirtualMachine lifecycle flows.
+
+This design proposal aims to set out new configurable behavior where any referenced instance types and/or preferences are expanded into the VirtualMachine during submission.
+
+## Motivation
+
+TBD
+
+## Goals
+
+* Provide simple configurables at the cluster and VirtualMachine level to expand instance types and preferences during submission
+
+## Non Goals
+
+* The default behavior will not change as part of this work
+
+## User Stories
+
+* As a cluster owner I want to expand all new VirtualMachines referencing instance types and preferences during submission  
+* As a VirtualMachine owner I want to optionally expand my VirtualMachine referencing an instance type or preference
```

Comment:
I think both user stories should state that it is an option that can be enabled?

> As a cluster owner I want to configure whether all new VirtualMachines referencing instance types and preferences are expanded during submission

> As a VirtualMachine owner I want to choose whether my VirtualMachine referencing an instance type or preference is expanded during submission